docs/api/components/schemas.yaml

```diff
@@ -3147,6 +3147,10 @@ IndexSettings:
       description: The number of partitions to use for the large indexes created by Sync Gateway. It is not recommended to set this unless you require additional horizontal scalability for individual indexes and have appropriately scaled your Query nodes to handle the increased query parallelism. If set, the recommended number is 8 and does not need to be directly related to the number of your Query nodes. Ensure documentation is read to understand the performance tradeoffs and instructions for migration if you have previously run with only one partition. See [/{db}/_index_init](#operation/post_db-_index_init) for more information.
       type: number
       default: 1
+    separate_principal_indexes:
+      description: Whether to create separate indexes for users and roles instead of a single larger syncDocs index.
```

Comment:
We should document this behavior. I think the default is do not do anything, but if there are already separate principal indexes.

Thinking more about this, we don't need to support going the opposite direction, so maybe this should just be called `create_separate_principal_indexes` if that is simpler. And then it can no-op if you call it again.

The num_partitions does not check the existing db state and I don't know if it's helpful to have this be restricted more strongly?

Reply:
It doesn't check db state but it does check if the existing config has a matching partition count

```go
	currentDbConfig := h.server.GetDatabaseConfig(h.db.Name)
	if req.NumPartitions != nil && currentDbConfig.NumIndexPartitions() == *req.NumPartitions {
		return base.HTTPErrorf(http.StatusBadRequest, "num_partitions is already %d", *req.NumPartitions)
	}
```

I assumed `UseLegacySyncDocsIndex()` would be based on the existence of an index on startup but maybe that's wrong - there's not really an equivalent for this option AFAIK, since we're not planning on exposing it to `DbConfig`?